src/cowboy_http.erl

```diff
@@ -75,6 +75,8 @@
 	version = undefined :: cowboy:http_version(),
 	%% Unparsed te header. Used to know if we can send trailers.
 	te :: undefined | binary(),
+    %% Body remaining to be sent when streaming response.
+    body_remaining :: non_neg_integer(),
```

Comment:
I wasn't sure if `body_remaining` should be added to the stream or state records.

Reply:
Stream is good. However I would rather have it mirror something http2 already does for incoming requests: keep around the expected size and the size sent rather than just the size remaining, because that info makes it much easier to debug things. So in this case it'd be fields `local_expected_size` and `local_sent_size`, the latter incremented before each send/sendfile and it needs to be checked also before sending because we don't want to ever send more than advertised.